lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -879,7 +879,7 @@ def parse_subspecs(name, requirements)
         end if test_specs
 
         requirements.pop if options.empty?
-        !subspecs.nil? || !test_specs.nil?
```

Comment:
We should not include test specs here because if this method returns true then only the test specs are being parsed.